pipeline/inputs/prometheus-remote-write.md

```diff
@@ -0,0 +1,74 @@
+---
+description: An input plugin to ingest payloads of Prometheus remote write
+---
+
+# Prometheus Remote Write
+
+This input plugin allows you to ingest a payload in the Prometheus remote-write format, i.e. a remote write sender can transmit data to Fluent Bit.
+
+## Configuration
+
+| Key           | Description                                                                                                                                    | default |
+| ----------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | ------- |
+| listen            | The address to listen on                                                                                                                       | 0.0.0.0 |
+| port              | The port for Fluent Bit to listen on                                                                                                           | 8080    |
+| buffer\_max\_size   | Specify the maximum buffer size in KB to receive a JSON message.                                                                               | 4M      |
+| buffer\_chunk\_size | This sets the chunk size for incoming incoming JSON messages. These chunks are then stored/managed in the space available by buffer_max_size.  | 512K    |
+|successful\_response\_code | It allows to set successful response code. `200`, `201` and `204` are supported.| 201 |
+| tag\_from\_uri      | If true, tag will be created from uri, e.g. api\_prom\_push from /api/prom/push, and any tag specified in the config will be ignored. If false then a tag must be provided in the config for this input. | true    |
+
+
+
+A sample config file to get started will look something like the following:
+
+
+{% tabs %}
+{% tab title="fluent-bit.conf" %}
+```
+[INPUT]
+	name prometheus_remote_write
+	listen 127.0.0.1
+	port 8080
+	uri /api/prom/push
+
+[OUTPUT]
+	name stdout
+	match *
+```
+{% endtab %}
+
+{% tab title="fluent-bit.yaml" %}
+```yaml
+pipeline:
+    inputs:
+        - name: prometheus_remote_write
+          listen: 127.0.0.1
+          port: 8080
+          uri: /api/prom/push
+    outputs:
+        - name: stdout
+          match: '*'
+```
+{% endtab %}
+{% endtabs %}
+
+With the above configuration, Fluent Bit will listen on port `8080` for data. You can now send payloads of Prometheus remote write to the endpoints `/api/prom/push`.
+
+## Examples
+
+### Communicate with TLS
+
+Communicating with TLS, you will need to use the tls related parameters:
```

Comment:
I'd link out to the transport-security doc as well.

Reply:
I added a link for it and added mention this plugin there: https://github.com/fluent/fluent-bit-docs/pull/1363/commits/0c1b1999c2caa63adf0303933bf912a1bc1f9c4a